packages/world/src/utils/ForceFieldUtils.sol

```diff
@@ -90,12 +91,10 @@ library ForceFieldUtils {
    */
   function removeFragment(EntityId forceField, EntityId fragment) internal {
     require(!fragment._getProgram().exists(), "Can't remove a fragment with a program");
-
-    (EnergyData memory machineData,) = updateMachineEnergy(forceField);
-    Energy._setDrainRate(
-      forceField, machineData.drainRate - MACHINE_ENERGY_DRAIN_RATE - Fragment._getExtraDrainRate(fragment)
-    );
-    Fragment._deleteRecord(fragment);
```

Comment:
Why dont we delete record anymore and instead set forcefield id to null?

Reply:
because we shouldn't remove the extra drain rate 😞 I'm hoping this hadn't happened yet, as it is kind of a rare edgecase at this point in the game, but we will need to run a migration if it happened.

Reply:
will add an issue to track